File: unity/Assets/Scripts/Multiplayer/Players/PlayerAuthResponse.cs
using Michitai.Multiplayer.Errors;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Players
{
    /// <summary>
    /// Response returned when a player is successfully authenticated.
    /// Contains the player's full information including typed player data.
    /// </summary>
    /// <typeparam name="T">The type to deserialize player data into.</typeparam>
    [System.Serializable]
    public class PlayerAuthResponse<T> : ApiResponse<EPlayerLoginError> where T : class, new()
    {
        /// <summary>
        /// The authenticated player's information including ID, name, and custom data.
        /// </summary>
        public PlayerInfo<T> player;
    }
}

File: java/src/main/java/com/michitai/multiplayer/players/PlayerAuthResponse.java
package com.michitai.multiplayer.players;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.michitai.multiplayer.ApiResponse;

/**
 * Response returned when a player is successfully authenticated.
 * Contains the player's full information including typed player data.
 *
 * @param <T> The type to deserialize player data into.
 */
public class PlayerAuthResponse<T> extends ApiResponse {
    @JsonProperty("player")
    private PlayerInfo<T> player;

    public PlayerInfo<T> getPlayer() {
        return player;
    }

    public void setPlayer(PlayerInfo<T> player) {
        this.player = player;
    }
}
